django/db/migrations/autodetector.py

```diff
@@ -1304,19 +1302,20 @@ def suggest_name(cls, ops):
         represent. Names are not guaranteed to be unique, but put some effort
         into the fallback name to avoid VCS conflicts if possible.
         """
-        if len(ops) == 1:
-            if isinstance(ops[0], operations.CreateModel):
-                return ops[0].name_lower
-            elif isinstance(ops[0], operations.DeleteModel):
-                return "delete_%s" % ops[0].name_lower
-            elif isinstance(ops[0], operations.AddField):
-                return "%s_%s" % (ops[0].model_name_lower, ops[0].name_lower)
-            elif isinstance(ops[0], operations.RemoveField):
-                return "remove_%s_%s" % (ops[0].model_name_lower, ops[0].name_lower)
-        elif ops:
-            if all(isinstance(o, operations.CreateModel) for o in ops):
-                return "_".join(sorted(o.name_lower for o in ops))
-        return "auto_%s" % get_migration_name_timestamp()
+        if not ops:
+            return "empty"
+
+        name = ""
+        for op in ops:
```

Comment:
Maybe this could be:

```python
name = ops[0].suggest_migration_name()
suggested_names = (op.suggest_migration_name() for op in ops[1:])
for suggested_name in suggested_names:
    name += f'_{suggested_name}'
    if len(name) > 42:
        break
```
To me at least, it reads a bit easier.

Reply:
Sure, forgot we can use f-strings on master 😸 Have adjusted slightly to use a generator for all and always break before the limit. Also your code fixes a bug where the name would be empty if the first suggested name is >42 characters.